transport/server.cc

```diff
@@ -962,26 +962,19 @@ make_result(int16_t stream, messages::result_message& msg, const tracing::trace_
 template<typename Process>
 requires std::same_as<std::invoke_result_t<Process, service::client_state&, distributed<cql3::query_processor>&, request_reader,
         uint16_t, cql_protocol_version_type, service_permit, tracing::trace_state_ptr, bool, cql3::computed_function_values>, future<cql_server::process_fn_return_type>>
-future<cql_server::result_with_foreign_response_ptr>
```

Comment:
> Re-bounce the msg to another shard if needed,
> e.g. in the case of tablet migration.

Please give an example of how tablet migration requires this.

Reply:
okay.  @tgrabiec if you have an example ready I'll be happy to adopt it.

Reply:
Bouncing happens when executing LWT statement in `modification_statement::execute_with_condition` by returning a special result message kind. The code assumes that after jumping to the shard from the bounce request, the result message is the regular one and not yet another bounce. There is no problem with vnodes, because shards don't change. With tablets, they can change at run time on migration.

Reply:
Added the above to commitlog in v4